content/08-cr-features.md

```diff
@@ -585,6 +585,54 @@ Whether a loop is unrolled or not depends on the loop unrolling cost, which is d
 
 
 
+<!--label:cr-tail-recursion-elimination-->
+### Tail Recursion Elimination (TRE)
+
+##### since Haxe 4.1.0
+
+Recursion is a natural way to solve some tasks like calculating fibonacci numbers or traversing trees. It's easy to write,
+read and understand. But it takes more computational resources at run time than solving the same tasks with loops instead of recursion.
+
+For example a loop-based function to find the root of a tree of nodes:
+```haxe
+  static function getRoot(node:Node):Node {
+    var root = node;
+    while(root.parent != null) {
+      root = root.parent;
+    }
+    return root;
+  }
+```
+Its recursive counterpart may look cleaner, but has worse performance because of additional function calls:
```

Comment:
I guess it's better to be a bit vague here and say "might have" instead of "has".